libs/ui/design-system/src/lib/copy-wrapper.tsx

```diff
@@ -0,0 +1,112 @@
+'use client'
+
+import { CheckCircle, Copy } from 'lucide-react'
+import { type FC, type PropsWithChildren, useRef, useState } from 'react'
+import { toast } from 'sonner'
+
+type Position = 'top-right' | 'top-left' | 'bottom-right' | 'bottom-left'
+
+interface CopyWrapperProps {
+  showCopyButton?: boolean
+  position?: Position
+  onCopySuccess?: () => void
+  onCopyError?: (error: Error | string) => void
+  className?: string
```

Comment:
I believe Mixins would be a great way to compose the props in the future.

Reply:
Yes, but... it should be done in one iteration everywhere. :)

Reply:
Sounds good